File: java/src/main/java/sunnyland/Cursor.java
package sunnyland;

public class Cursor extends GameObject {
    public Cursor() {
        super();
    }

    public void mousePos(double mouseX, double mouseY, int winWidth, int winHeight) {
        pos.x = (float) ((2.0f * mouseX) / winWidth - 1.0f);
        pos.y = (float) (1.0f - (2.0f * mouseY) / winHeight);
        setPosition(pos.x, pos.y);
    }
}

File: cpp/src/Cursor.cpp
#include "cursor.h"

void Cursor::mousePos(double mouseX, double mouseY, int winWidth, int winHeight) {
    pos.x = static_cast<float>((2.0f * mouseX) / winWidth - 1.0f);
    pos.y = static_cast<float>(1.0f - (2.0f * mouseY) / winHeight);
    setPosition(pos.x, pos.y);
}
